src/components/AnalysisName.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Button, Input } from 'semantic-ui-react'
+import * as app from '../store/actions/app'
+
+class AnalysisName extends React.Component {
+  static propTypes = {
+    dispatch: PropTypes.func.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      isEditing: false
+    }
+
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleClick = this.handleClick.bind(this)
+    this.handleCancel = this.handleCancel.bind(this)
+  }
+
+  handleSubmit (event) {
+    let input = this.refs.viewName.inputRef.value
+    if (input === '') {
+      input = 'Untitled Analysis'
+    }
+    this.setState({ isEditing: false })
+    document.title = input + ' | OpenTraffic Analyst UI'
+    this.props.dispatch(app.setAnalysisName(input))
+  }
+
+  handleClick (event) {
+    this.setState({ isEditing: true })
+  }
+
+  handleCancel (event) {
+    this.setState({ isEditing: false })
+  }
+
+  render () {
+    if (this.state.isEditing) {
+      return (
+        <div className="editText">
+          <form onSubmit={this.handleSubmit}>
+            <Input type="text" action ref="viewName">
+              <input defaultValue={this.props.viewName} />
```

Comment:
One other thing that would be really handy to do is to do a `.focus()` and maybe a `.select()` on the input box contents, when it's activated. (You might have to do it in a `componentDidUpdate()` handler.) This means when someone edits the box